---
title: "Enrichment Map Workflow"
author: "Ruth Isserlin"
date: '`r format(Sys.Date())`'
output:
  html_document:
    theme: paper
    toc: yes
---

```{r, echo=FALSE}
htmltools::img(src = knitr::image_uri(file.path(getwd(),"Figures" ,"enrichmentmap_logo_mini.png")), 
               alt = 'logo', 
               style = 'position:absolute; top:0; right:0; padding:65px;')
```



Set of R notebooks to transform expression data to a ranked list and run them through Pathay enrichment pipeline. Pathway enrichment analysis helps gain mechanistic insight into large gene lists typically resulting from genome scale (â€“omics) experiments. It identifies biological pathways that are enriched in the gene list more than expected by chance. We explain pathway enrichment analysis and present a practical step-by-step guide to help interpret gene lists. The protocol is designed for biologists with no prior bioinformatics training and uses freely available software including g:Profiler, GSEA, Cytoscape and Enrichment Map.

The main protocol involves translating a ranked list to an Enrichment map. The figure below outlines the main steps.


![EM Protocol Flow Chart](./Figures/Figure1_Protocol_Flow_Chart.png)

# Table of Contents

 1. [Download TCGA data](Download_TCGA_data.html) - R notebook shows you how to download legacy microarray and rnaseq ovarian cancer data. The notebook can be modified to download any data from GDC
 1. [Supplemental Protocol 1](supplemental_protocol1_rnaseq.html) - convert raw RNASeq expression data to a ranked list
 1. [Supplemental Protocol 2](supplemental_protocol2_microarray.html) - convert RMA normalized microarray expression data to a ranked list
 1. [Supplemental Protocol 3](supplemental_protocol3_R_gse_methods.html) - perform Pathway Enrichment Analysis in R using ROAST and Camera
 1. [Supplemental Protocol 4](Supplemental_protocol_4_manual_phenotype_rand_with_edgeR.html) - perform phenotype randomizations using edgeR with GSEA.
 1. [Supplemental Protocol 5](Supplementary_Protocol5_Multi_dataset_theme_analysis.html) - Creates a Multi dataset Enrichment Map (using GSEA results from multiple rank files) and followed by Theme calculation and summary.
 1. [Protocol 2](Protocol2_createEM.html) - run GSEA on ranked list and automatically create an Enrichment Map from the results.
 1. [Protocol 2 add on](Protocol2_annotateEM.html) - take the resulting EM from Protocol 2 and annotate it.  Pull all the annotations directly into R to be used for further analysis.
 
# Data Files used for the Protocol

 1. [Supplmentary table 1](/Cytoscape_workflows/EnrichmentMapPipeline/SupplementaryData/Supplementary_Table1_Cancer_drivers.txt) - Cancer drivers.  List of genes used as g:Profiler input.
 1. [Supplmentary table 2](/Cytoscape_workflows/EnrichmentMapPipeline/SupplementaryData/Supplementary_Table2_MesenvsImmuno_RNASeq_ranks.rnk) - GSEA Rank file.  List of genes and their assoicated ranks used as input for GSEA.
 1.  [Supplmentary table 3](/Cytoscape_workflows/EnrichmentMapPipeline/SupplementaryData/Supplementary_Table3_Human_GOBP_AllPathways_no_GO_iea_July_01_2017_symbol.gmt) - GSEA GMT file.  Pathway defintion file used for the protocol as GSEA input.  Updated gmt files can be found on the baderlab download site [here](http://download.baderlab.org/EM_Genesets/current_release/)
 1. [Supplmentary table 4](/Cytoscape_workflows/EnrichmentMapPipeline/SupplementaryData/Supplementary_Table4_gprofiler_results.txt) - Example g:Profiler results file. 
 1. [Supplmentary table 5](/Cytoscape_workflows/EnrichmentMapPipeline/SupplementaryData/Supplementary_Table5_hsapiens.pathways.NAME.gmt) - Example g:Profiler gmt file. If using example g:Profiler results file instead of regenerating the results make sure to use this gmt file.  If the g:Profiler results file has been regenerated then make sure to download an updated gmt file that corresponds to the results file as specified in the protocol.  
 1. [Supplmentary table 6](/Cytoscape_workflows/EnrichmentMapPipeline/SupplementaryData/Supplementary_Table6_TCGA_OV_RNAseq_expression.txt) - Example expression file (corresponding to example rank file, supplementary 2) used when creating an Enrichment Map from GSEA results.  
 1. [Supplmentary table 7](/Cytoscape_workflows/EnrichmentMapPipeline/SupplementaryData/Supplementary_Table7_TCGA_OV_RNAseq_classes.cls) - Example class file (corresponding to example expression file, supplementary 6) used when creating an Enrichment Map from GSEA results.
 1. [Supplmentary table 8](/Cytoscape_workflows/EnrichmentMapPipeline/SupplementaryData/Supplementary_Table8_gsea_report_for_na_pos.xls) - Example GSEA results file 1. 
 1. [Supplmentary table 9](/Cytoscape_workflows/EnrichmentMapPipeline/SupplementaryData/Supplementary_Table9_gsea_report_for_na_neg.xls) - Example GSEA results file 2. 
 1. [Supplmentary table 10](/Cytoscape_workflows/EnrichmentMapPipeline/SupplementaryData/Supplementary_Table10_TCGA_RNASeq_rawcounts.txt) - Example RANSeq expression input file used in Supplementary Protocol 1.  File used to generate rank file, supplementary table 2, used in the main protocol. 
 1. [Supplmentary table 11](/Cytoscape_workflows/EnrichmentMapPipeline/SupplementaryData/Supplementary_Table11_RNASeq_classdefinitions.txt) - Example sample definition input file used in Supplementary Protocol 1.  File used to generate rank file, supplementary table 2, used in the main protocol.  
 1. [Supplmentary table 12](/Cytoscape_workflows/EnrichmentMapPipeline/SupplementaryData/Supplementary_Table12_TCGA_Microarray_rmanormalized.txt) - Example Microarray expression input file used in Supplementary Protocol 2.  
 1. [Supplmentary table 13](/Cytoscape_workflows/EnrichmentMapPipeline/SupplementaryData/Supplementary_Table13_Microarray_classdefinitions.txt) - Example sample definition input file used in Supplementary Protocol 2.  
 
 Download all the above files [here](/Cytoscape_workflows/EnrichmentMapPipeline/SupplementaryData_EM_Protocol.tar.gz)
    
  See the [GitHub repository](https://github.com/BaderLab/Cytoscape_workflows) for the source.
